src/app/gateways/views/GatewayListTabsView.vue

```diff
@@ -11,56 +11,56 @@
         mesh: '',
       }"
     >
-      <AppView>
-        <template #title>
-          <h2>
-            <RouteTitle :title="t(`${currentRoute.name === 'builtin-gateway-list-view' ? 'builtin' : 'delegated'}-gateways.routes.items.title`)" />
-          </h2>
-        </template>
-
-        <template #actions>
-          <LinkBox>
-            <RouterLink
-              :class="{
-                'active': currentRoute.name === 'builtin-gateway-list-view',
-              }"
-              :to="{
-                name: 'builtin-gateway-list-view',
-                params: {
-                  mesh: route.params.mesh,
-                },
-              }"
-            >
-              {{ t('gateways.routes.items.navigation.builtin') }}
-            </RouterLink>
-
-            <RouterLink
-              :class="{
-                'active': currentRoute.name === 'delegated-gateway-list-view',
-              }"
-              :to="{
-                name: 'delegated-gateway-list-view',
-                params: {
-                  mesh: route.params.mesh,
-                },
-              }"
+      <RouterView
+        v-slot="{ Component, route: active }"
+      >
```

Comment:
**Question**: Why use this instead of keeping to checking `currentRoute.name`? This seems unnecessarily verbose.

Reply:
I understood you are referring to the bit I removed here?

```js
import { useRoute } from 'vue-router'
...
...
const currentRoute = useRoute()
```

Yeah, I suppose that ^ seems more verbose to me than `{ route: active }`, and especially unnecessary doing all the import/using if its already exported from `RouterView`?

Reply:
Except that this made it necessary to move the RouterView all the way up the tree and hook up its `<component ...>` bit, too. The diff for this PR could’ve been much smaller without this change. Also, it’s quite strange to say that needing to import something is verbose. How can an import be verbose? Never mind that the import was already here to begin with.

Imports are fine and good. It’s not needed to go out of your way and remove them, especially if that requires changing a bunch of other stuff in otherwise working code.